bench/BenchAll.hs

```diff
@@ -415,6 +415,8 @@ main = do
           nf (S.foldl' (\acc x -> acc + fromIntegral x) (0 :: Int)) s) foldInputs
       , bgroup "foldr'" $ map (\s -> bench (show $ S.length s) $
           nf (S.foldr' (\x acc -> fromIntegral x + acc) (0 :: Int)) s) foldInputs
+      , bgroup "unfoldrN" $ map (\s -> bench (show $ S.length s) $
+          nf (S.unfoldrN (S.length s)) (\a -> Just (a, a + 1)) ) foldInputs
```

Comment:
There appears to be something amiss with these benchmarks. When I run them the output looks like this:

```
    unfoldrN
      1:     OK (0.18s)
         17 ns ± 1.3 ns
      2:     OK (0.16s)
         18 ns ± 1.5 ns
      4:     OK (0.30s)
         17 ns ± 754 ps
...
```

The runtime is the same for each input length, varying only slightly with different compiler versions.

Any ideas what's going on?

Reply:
https://github.com/haskell/bytestring/pull/356#discussion_r573140330

Reply:
The closure was simply missing an argument! :)

```suggestion
      , bgroup "unfoldrN" $ map (\s -> bench (show $ S.length s) $
          nf (S.unfoldrN (S.length s) (\a -> Just (a, a + 1))) 0) foldInputs
```